src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -4022,4 +4083,463 @@ public void SyntaxTreeAlreadyPresent()
+            // partial class C : FI
+            Diagnostic(ErrorCode.ERR_UnimplementedInterfaceMember, "FI").WithArguments("C", "FI.M()").WithLocation(6, 19));
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/68219")]
+    public void PartialExplicitImplementation_03()
+    {
+        var source1 = """
+            using System.Collections.Generic;
+
+            file interface IFoo
+            {
+                IReadOnlyDictionary<int, string> Bar { get; }
+            }
+
+            internal partial class Foo : IFoo
+            {
+                private readonly Dictionary<int, string> _bar = new() { { 1, "one" }, { 2, "two" } };
+                IReadOnlyDictionary<int, string> IFoo.Bar => _bar;
+            }
+            """;
+
+        var source2 = """
+            using System.Collections.Generic;
+
+            file interface IFoo
+            {
+                IReadOnlyDictionary<int, string> Bar { get; }
+            }
+
+            internal partial class Foo : IFoo
+            {
+                IReadOnlyDictionary<int, string> IFoo.Bar => _bar;
+            }
+            """;
+
+        var comp = CreateCompilation(new[] { (source1, "F1.cs"), (source2, "F2.cs") });
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/68219")]
+    public void PartialExplicitImplementation_04()
+    {
+        var source1 = """
+            file interface IFoo
+            {
+                int Bar { get; }
+            }
+
+            internal partial class Foo : IFoo
+            {
+            }
+            """;
+
+        var source2 = """
+            file interface IFoo
+            {
+                int Bar { get; }
+            }
+
+            internal partial class Foo : IFoo
+            {
+                int IFoo.Bar => 1;
+                int IFoo.Bar => 2;
+            }
+            """;
+
+        var comp = CreateCompilation(new[] { (source1, "F1.cs"), (source2, "F2.cs") });
+        comp.VerifyDiagnostics(
+            // F1.cs(6,24): error CS8646: 'IFoo.Bar' is explicitly implemented more than once.
+            // internal partial class Foo : IFoo
+            Diagnostic(ErrorCode.ERR_DuplicateExplicitImpl, "Foo").WithArguments("IFoo.Bar").WithLocation(6, 24),
+            // F1.cs(6,30): error CS0535: 'Foo' does not implement interface member 'IFoo.Bar'
+            // internal partial class Foo : IFoo
+            Diagnostic(ErrorCode.ERR_UnimplementedInterfaceMember, "IFoo").WithArguments("Foo", "IFoo.Bar").WithLocation(6, 30),
+            // F2.cs(9,14): error CS0102: The type 'Foo' already contains a definition for '<F2>F141A34209AF0D3C8CA844A7D9A360C895EB14E557F17D27626C519D9BE96AF4A__IFoo.Bar'
+            //     int IFoo.Bar => 2;
+            Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "Bar").WithArguments("Foo", "<F2>F141A34209AF0D3C8CA844A7D9A360C895EB14E557F17D27626C519D9BE96AF4A__IFoo.Bar").WithLocation(9, 14));
```

Comment:
It's not ideal to expose the file type's metadata name in an error message, but it's also not a big problem.

If we decide on some way in the future to produce smaller/simpler mangled names for these types, it may be helpful in more places. For example, it's not easy to skim such type names when looking through an ETL trace.